common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1118,12 +1122,21 @@ def _block_matches_all(block_data):
         if 'category' in qualifiers:
             qualifiers['block_type'] = qualifiers.pop('category')
 
+        # set `in_root_only` feature if 'in_root_only' kwarg is present
+        in_root_only = kwargs.get('in_root_only', False)
+
         # don't expect caller to know that children are in fields
         if 'children' in qualifiers:
             settings['children'] = qualifiers.pop('children')
         for block_id, value in course.structure['blocks'].iteritems():
             if _block_matches_all(value):
-                items.append(block_id)
+                # If `in_root_only` feature enabled
+                if in_root_only:
+                    # Check if the block is in the course tree
+                    if self.has_path_to_root(block_id, course):
```

Comment:
I also realized that there are some "detached" blocks, like `CouseInfo` that won't have a `path_to_root` but we still want to be returned in `get_items`. So you'll also want to check for `detached` blocks, like we do here: https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/split_mongo/split.py#L1192